### PR TITLE
[fix] Hide Octospinner when opening in a new tab/new window.

### PR DESCRIPTION
**Pull request by dinever**
Hi guys,

This is a fix to #442.

Any feedback would be greatly appreciated.

Thanks! :smile:

### CODE REVIEW

File: app/assets/javascripts/navigation.coffee [hunk 1]
```diff
@@ -1,5 +1,12 @@
 ready = ->
-  $('.js-navigation').on('click', -> $('.loading-indicator').show())
+  $('.js-navigation').on('click', (event) ->
+    if (event.ctrlKey or # New tab (windows)
+    event.shiftKey or # New window
+    event.metaKey or # New tab (mac)
+    (event.button and event.button == 1) # New tab (middle click)
+    )
+      return
+    return $('.loading-indicator').show())
```
Comment (tarebyte) [inline app/assets/javascripts/navigation.coffee]: So this is a good start, and I like the documentation, but what do you think about something like this?

I think extracting it out into it's own method makes it a little more readable.

Just a style note I'm a fan of guard clauses

```coffee
return if foo
# continue on
```

Also coffeescript much like Ruby the last line of the method is returned as the result. So the last result is not necessary.

```coffee
# A new window/tab event can be one of the following
# ctrlKey (Windows)
# metaKey (OSX)
# shiftKey (OS agnositc)
# New tab with middle click
new_window_or_tab_event = (event) ->
  middle_click_event = (event.button && event.button == 1)
  event.ctrlKey || event.metaKey || event.shiftKey || middle_click_event

ready = ->
  $('.js-navigation').on('click', (event) ->
    return if new_window_or_tab_event(event)
    $('.loading-indicator').show())
```

**Comment by dinever** [comment]
Hi @tarebyte,

You are right. I had the style improved.

Thanks!

**Comment by johndbritton** [comment]
This looks good to me. Thanks for the contribution @dinever.

@tarebyte If you verified this, I say go for it. Maybe file another issue about the Turbolinks 3 progress bar / spinner you wanted to upgrade to for the future.

**Comment by tarebyte** [comment]
I can confirm this works locally, thanks @dinever

**Comment by johndbritton** [comment]
> This is a fix to #442.

@dinever Just an FYI, if you use the term `Fixes #442`, `Closes #442` or similar in your pull request, the issue will automatically be closed when the pull request is merged. https://github.com/blog/1506-closing-issues-via-pull-requests

**Comment by dinever** [comment]
Thanks @johndbritton, I'll remember that.

**Pull request merged**